stubs/bleach/bleach/html5lib_shim.pyi

```diff
@@ -1,27 +1,59 @@
-from _typeshed import Incomplete
-from collections.abc import Generator, Iterable
+import re
+from collections.abc import Iterable, Iterator
+from typing import Any
 
-class HTMLParser:  # actually html5lib.HTMLParser
-    def __getattr__(self, __name: str) -> Incomplete: ...
+# We don't re-export any `html5lib` types / values here, because they are not
+# really public and may change at any time. This is just a helper module,
+# import things directly from `html5lib` instead!
+from html5lib import HTMLParser
+from html5lib._inputstream import HTMLUnicodeInputStream
+from html5lib._tokenizer import HTMLTokenizer
+from html5lib.serializer import HTMLSerializer
 
-class Filter:  # actually html5lib.filters.base.Filter
-    def __getattr__(self, __name: str) -> Incomplete: ...
+HTML_TAGS: list[str]
+HTML_TAGS_BLOCK_LEVEL: frozenset[str]
+AMP_SPLIT_RE: re.Pattern[str]
+ENTITIES: dict[str, str]
+TAG_TOKEN_TYPES: set[int]
+TAG_TOKEN_TYPE_CHARACTERS: int
+TAG_TOKEN_TYPE_END: int
+TAG_TOKEN_TYPE_PARSEERROR: int
+TAG_TOKEN_TYPE_START: int
 
-class SanitizerFilter:  # actually html5lib.filters.sanitizer.Filter
-    def __getattr__(self, __name: str) -> Incomplete: ...
+class InputStreamWithMemory:
+    position: int
+    def __init__(self, inner_stream: HTMLUnicodeInputStream) -> None: ...
+    def reset(self) -> None: ...
+    @property
+    def errors(self) -> list[str]: ...
+    @property
+    def charEncoding(self) -> tuple[str, str]: ...
+    # Is a property returning a method, simplified:
+    def changeEncoding(self, newEncoding: str) -> None: ...
+    def char(self) -> str: ...
+    def charsUntil(self, characters: str, opposite: bool = ...) -> str: ...
```

Comment:
The bleach method just delegates to `HTMLUnicodeInputStream.charsUntil`, and [the docstring of _that_ method](https://github.com/html5lib/html5lib-python/blob/f7cab6f019ce94a1ec0192b6ff29aaebaf10b50d/html5lib/_inputstream.py#L314-L319) states:

> 'characters' must be a container that supports the 'in' method and iteration over its characters.

```suggestion
    def charsUntil(self, characters: Collection[str], opposite: bool = ...) -> str: ...
```

(We could probably just go with `Iterable[str]`, but `Collection[str]` seems safer, since it guarantees the object will have a `__contains__` method.)

Reply:
I hopped in their Matrix chat and asked for clarification. In the mean time let's use Collection to respect the contract established by the docstring as-written, even if I think it just meant to be iterable.

Reply:
@AlexWaygood I got a response
![image](https://github.com/python/typeshed/assets/1350584/61000800-3b71-4299-b3a9-d0355929c291)
https://matrix.to/#/!AGetWbsMpFPdSgUrbs:matrix.org/$_Vh0jaWtF_F60H5k28DZzVtIGKnG--lWc-vDrLvhUrI?via=matrix.org&via=mozilla.org&via=envs.net

Reply:
Happy to go with `Iterable[str]` if you prefer!